src/main/java/openperipheral/addons/sensors/AdapterSensor.java

```diff
@@ -47,6 +49,85 @@ private SupportedEntityTypes(Class<? extends Entity> cls) {
 		}
 	}
 
+	private static enum ComputerColors {
+		WHITE     (0xF0F0F0, "0"),
+		ORANGE    (0xF2B233, "1"),
+		MAGENTA   (0xE57FD8, "2"),
+		LIGHT_BLUE(0x99B2F2, "3"),
+		YELLOW    (0xDEDE6C, "4"),
+		LIME      (0x7FCC19, "5"),
+		PINK      (0xF2B2CC, "6"),
+		GRAY      (0x4C4C4C, "7"),
+		LIGHT_GRAY(0x999999, "8"),
+		CYAN      (0x4C99B2, "9"),
+		PURPLE    (0xB266E5, "a"),
+		BLUE      (0x3366CC, "b"),
+		BROWN     (0x7F664C, "c"),
+		GREEN     (0x57A64E, "d"),
+		RED       (0xCC4C4C, "e"),
+		BLACK     (0x000000, "f");
+
+		public final int R;
+		public final int G;
+		public final int B;
+		public final String code;
+
+		// LRU cache with a maximum of 1000 entries
+		private static final LinkedHashMap<Integer, ComputerColors> LOOKUP_CACHE = new LinkedHashMap<Integer, ComputerColors>(16, 0.75f, true) {
+		  protected boolean removeEldestEntry(Map.Entry<Integer, ComputerColors> eldest) {
+		    return size() > 1000;
+		  }
+		};
+
+		ComputerColors(int rgb, String code) {
+			this.R = (rgb >>> 16) & 0xFF;
+			this.G = (rgb >>> 8) & 0xFF;
+			this.B = rgb & 0xFF;
+			this.code = code;
+		}
+
+		// Formula taken from http://www.compuphase.com/cmetric.htm
+		private int distanceSquared(int cr, int cg, int cb) {
+			int r_mean = (this.R + cr) >>> 1;
+			int r_delta = this.R - cr;
+			int g_delta = this.G - cg;
+			int b_delta = this.B - cb;
+
+			int rd2 = r_delta*r_delta;
+			int gd2 = g_delta*g_delta;
+			int bd2 = b_delta*b_delta;
+
+			return Math.abs((((512+r_mean)*rd2)>>8)
+					   + (gd2 << 2)
+					   + (((767-r_mean)*bd2)>>8));
+		}
+
+		public static ComputerColors getClosestColor(int rgb) {
+			ComputerColors cached = ComputerColors.LOOKUP_CACHE.get(rgb);
+			if (cached != null) {
+				return cached;
+			}
+
+			int cr = (rgb >>> 16) & 0xFF;
+			int cg = (rgb >>> 8) & 0xFF;
+			int cb = rgb & 0xFF;
+
+			ComputerColors lowest = null;
+			int lowest_score = Integer.MAX_VALUE;
+
+			for (ComputerColors cc : ComputerColors.values()) {
+				int cc_score = cc.distanceSquared(cr, cg, cb);
+				if (lowest == null || cc_score < lowest_score) {
+					lowest = cc;
+					lowest_score = cc_score;
+				}
+			}
+
+			ComputerColors.LOOKUP_CACHE.put(rgb, lowest);
+			return lowest;
+		}
+	}
+
```

Comment:
There is already something similar in https://github.com/OpenMods/OpenModsLib/blob/master/src/main/java/openmods/utils/ColorUtils.java. It has inferior "nearest" metric, but changing that was already on my improvements list. It also does not have color->hex functionality, though `bitmask` member contains same values ones on CC `colors` module, which I prefer.

Anyway, if possible, can you use that class? If I improve there, it will be fixed in all other projects.